Sources/Nimble/Matchers/PostNotification.swift

```diff
@@ -11,18 +15,27 @@ internal class NotificationCollector {
     }
 
     func startObserving() {
-        self.token = self.notificationCenter.addObserverForName(nil, object: nil, queue: nil) { [weak self] n -> Void in
-            self?.observedNotifications.append(n)
+        self.token = self.notificationCenter.addObserverForName(nil, object: nil, queue: nil) {
+            // linux-swift gets confused by .append(n)
+            [weak self] n in self?.observedNotifications += [n]
         }
     }
 
     deinit {
-        if let token = self.token {
-            self.notificationCenter.removeObserver(token)
-        }
+        #if _runtime(_ObjC)
+            if let token = self.token {
+                self.notificationCenter.removeObserver(token)
+            }
+        #else
+            if let token = self.token as? AnyObject {
+                self.notificationCenter.removeObserver(token)
+            }
+        #endif
     }
 }
 
+private let mainThread = pthread_self()
```

Comment:
Swift globals are always lazily initialized. Should we be referencing this somewhere where we are certain to be on the main thread to ensure that this works as expected?